### PR TITLE
HIVE-26433: StorageHandler authorization doesn't work with "METASTORE" db type

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

- Enhance the propagation of objectName to 'internal:metastore://' for STORAGEHANDLER_URI type of HivePrivilegeObject in case of hive.sql.database.type=METASTORE 

- Allows the policy definitions to be created independently from the actual implementation of the HMS database

### Why are the changes needed?

Enables the authorization, makes it working compared to the current state where it does not work at all 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

On a cluster

### CODE REVIEW

File: jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcStorageHandler.java [hunk 1]
```diff
@@ -99,7 +101,10 @@ public void configureInputJobProperties(TableDesc tableDesc, Map<String, String>
   @Override
   public URI getURIForAuth(Table table) throws URISyntaxException {
     Map<String, String> tableProperties = HiveCustomStorageHandlerUtils.getTableProperties(table);
-    String host_url = tableProperties.get(Constants.JDBC_URL);
+    DatabaseType dbType = DatabaseType.valueOf(
+      tableProperties.get(JdbcStorageConfig.DATABASE_TYPE.getPropertyName()));
+    String host_url = DatabaseType.METASTORE == dbType ?
+      "internal:metastore://" : tableProperties.get(Constants.JDBC_URL);
```
Comment (saihemanth-cloudera) [inline jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcStorageHandler.java:107]: All the storage handlers will have prefix in the Storage URL including jdbc storage handler. Can we make this hardcoded uri also to be in the same format? Essentially you would need to add "JDBC://" before the string "internal:metastore:/" (please take out extra "/" after "metastore:/")?

**Review by saihemanth-cloudera — CHANGES_REQUESTED** [review]

File: jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcStorageHandler.java (hunk 1 above)
Comment (deniskuzZ) [reply jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcStorageHandler.java:107]: With the current setup
````
'HivePrivilegeObject':{'type':STORAGEHANDLER_URI, 'dbName':null, 'objectType':STORAGEHANDLER_URI, 'objectName':internal:metastore:///null, 'columns':[], 'partKeys':[], 'commandParams':[], 'actionType':OTHER, 'owner':hive}
````
elements={storage-type=internal:metastore; storage-url=/null; }

Why do we need to remove extra "/" after "metastore?

Comment (deniskuzZ) [reply jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcStorageHandler.java:107]: should it be `jdbc:metastore://`?

Comment (saihemanth-cloudera) [reply jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcStorageHandler.java:107]: Yeah, having `jdbc://internal:metastore:/` would be consistent with other Storage URIs. But I'm ok with your above suggestion as well.

Comment (deniskuzZ) [reply jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcStorageHandler.java]: updated, please take a look

**Review by saihemanth-cloudera — COMMENTED** [review]
Changes looks good to me. Pending clean jenkins tests.

**Pull request merged**